### PR TITLE
MQTT5 support

### PR DESCRIPTION
**Pull request by sbSteveK**
Add MQTT5 support to aws-crt-python.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/__init__.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ class NativeResourceTest(unittest.TestCase):
 
     def setUp(self):
         NativeResource._track_lifetime = True
-        #init_logging(LogLevel.Trace, 'stderr')
+        # init_logging(LogLevel.Trace, 'stderr')
```
Comment (TwistedTwigleg) [inline test/__init__.py:28]: Extremely minor nitpick: Remove?

File: source/mqtt5_client.c [hunk 2]
```diff
@@ -0,0 +1,2041 @@
...
+    aws_mqtt5_transform_websocket_handshake_complete_fn *complete_fn;
+    void *complete_ctx;
+
+    /* Python bindings we created to wrap the native request */
+    /* Necessary? */
+    PyObject *request_binding_py;
+    PyObject *headers_binding_py;
+};
+
+static const char *s_capsule_name_ws_handshake_transform_data = "aws_ws_handshake_transform_data";
```
Comment (TwistedTwigleg) [inline source/mqtt5_client.c:593]: Extremely minor: Should this be with the other static constants near the top of the file?

File: source/mqtt5_client.c [hunk 3]
```diff
@@ -0,0 +1,2041 @@
...
+        if (PyErr_Occurred()) {
+            PyErr_WriteUnraisable(PyErr_Occurred());
+            goto cleanup;
+        }
+    }
+
+    result = PyObject_CallFunction(
+        metadata->callback,
+        "(Hiis#O)",
+        error_code,                                /* H */
```
Comment (TwistedTwigleg) [inline source/mqtt5_client.c:1417]: Extremely minor: Missing `/* i */`

**Review by TwistedTwigleg — APPROVED** [review]
I left comments on a few **extremely minor** things that I noticed, but none of them are pressing nor would they need adjusting prior to cutting a release.

File: test/__init__.py (hunk 1 above)
Comment (sbSteveK) [reply test/__init__.py:28]: This was here when I originally saw it and I think it's left in here for the purpose of our own internal testing. The change reflected here is just the addition of that space between the content and the line =P

**Pull request merged**